File: app/models/task.rb
# == Schema Information
#
# Table name: tasks
#
#  id          :integer          not null, primary key
#  description :string
#  priority    :string
#  list_id     :integer
#  created_at  :datetime         not null
#  updated_at  :datetime         not null
#

class Task < ActiveRecord::Base
  belongs_to :list
  validates :description, length: { in: 1..140 }
end

File: app/assets/javascripts/models/task.js
app.tasks.model = {
  all: [],
  new: function Task(id, description, priority) {
    this.description = description;
    this.priority = priority;
    this.id = id;
    var self = this;
    (function initialize() {
      app.tasks.model.all.push(self)
    }())
  }
}
